### PR TITLE
feat: Separate Latency Metrics [DT-479]

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
**What this PR does / why we need it**: 
* Separate latency metrics into HTTP, gRPC, and Outbound.

**JIRA ID**: [DT-479]

**Notes for your reviewer**:
* Bootstrap will be updated after this is merged.

[DT-479]: https://outreach-io.atlassian.net/browse/DT-479

### CODE REVIEW

File: pkg/trace/call.go [hunk 1]
```diff
@@ -93,12 +186,27 @@ func EndCall(ctx context.Context) {
...
+
+	if info.kind == metrics.CallKindExternal {
+		info.ReportOutboundLatency()
+	} else {
+		switch info.name { //nolint:exhaustive //Why: we only report latency metrics in this case on HTTP/gRPC call types.
+		case CallTypeHTTP:
+			info.ReportHTTPLatency()
+		case CallTypeGRPC:
+			info.ReportGRPCLatency()
+		}
```
Comment (vk-outreach) [inline pkg/trace/call.go:198]: You could probably add a `callType.reportLatency(info)` and move the switch over to there.

**Review by vk-outreach — APPROVED** [review]
LGTM

**Review by jaredallard — APPROVED** [review]
LGTM. Glad we're doing this

File: pkg/trace/call.go (hunk 1 above)
Comment (george-e-shaw-iv) [reply pkg/trace/call.go:198]: nice idea, i'll do that

**Pull request merged**